wlm/src/main/java/ai/djl/serving/wlm/LmiConfigRecommender.java

```diff
@@ -89,12 +89,20 @@ private static void setMpiMode(Properties lmiProperties) {
     }
 
     private static void setTensorParallelDegree(Properties lmiProperties) {
-        if (lmiProperties.containsKey("option.tensor_parallel_degree")) {
-            return;
-        }
-        String tpDegree = Utils.getenv("TENSOR_PARALLEL_DEGREE", "max");
+        String tpDegree =
+                lmiProperties.getProperty(
+                        "option.pipeline_parallel_degree",
+                        Utils.getenv("TENSOR_PARALLEL_DEGREE", "max"));
         int ppDegree =
                 Integer.parseInt(lmiProperties.getProperty("option.pipeline_parallel_degree"));
+
+        // If enable_lora=true, tp to "max" so that there will only be one WorkerThread per model.
+        boolean enableLora = Boolean.parseBoolean(lmiProperties.getProperty("option.enable_lora"));
+        if (enableLora) {
+            logger.info("enable_lora is set to true, overriding tensor_parallel_degree to max");
```

Comment:
I'd prefer that we throw an exception if customer provides a tp degree that does not equal "max" or the maximum number of devices.

Reply:
I made some changes. Instead of setting tp, setting maxWorkers=1. Because we really want is to have only one worker.